### PR TITLE
[VIP-Travel] Create a New Travel Page and Terms and Conditions Modal (NewDot)

### PR DESCRIPTION
**Pull request by cdOut**
### Details

Implementation of the main New Travel Page and the Terms and Conditions Modal for NewDot.

### Fixed Issues

$ https://github.com/Expensify/App/issues/37102
$ https://github.com/Expensify/App/issues/37103

### Tests

1. Make sure you are using an account with the travel beta enabled or in the `Permissions.ts` file set the `canUseAllBetas` function to always return true.
2. Press the `+` sign to go into the menu and then go into `Book travel`.
3. In the newly displayed modal / page press the `Book travel` button to go further into the flow.
4. On the Terms & Conditions page press the checkbox to accept and proceed with the flow by pressing the `Continue` button.
5. In the network tab check that a correct data package `AcceptSpontanaTerms` has been sent as a resolution of this flow.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/88325488/7b86dddb-bcda-4653-a7ed-848767cfd436

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/88325488/455b91f6-594e-416f-a1a0-cfb1139d2f6c

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/88325488/9a17246d-4810-429c-b00f-9ae1321311c7

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/88325488/fdde5173-ac80-40b7-af05-270cc44b3e5a

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/88325488/7ee5e8ab-d518-4810-a8f9-02343cbf67c2

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/88325488/2ccff230-7288-40b1-b0d7-5514aad404f5

</details>

### CODE REVIEW

**Comment by cdOut** [comment]
Info for engineers from the expert agency Software Mansion, please hold off the review until I fix all the merge issues with main.

**Comment by twisterdotcom** [comment]
@cdOut and @smelaa - so we had a bit of a rollercoaster discussion on the ToS, but we do still require the modal for accepting Terms, so that's great.

However, our marketing team have a few copy points: cc @jamesdeanexpensify 

> In Global Create, can we update it to `Book travel` (instead of `Book Travel`) to match the casing of the rest of the options?
> In the "My trips" section, can we update it to:
> - Book and manage your trips
> - Use Expensify Travel to get the best travel offers and manage your business expenses in one place.
> - Save money on your bookings
> - Get realtime updates and alerts
> [Button] Book travel

One last question in Slack too: https://expensify.slack.com/archives/C05S5EV2JTX/p1712789189082939?thread_ts=1712595510.349369&cid=C05S5EV2JTX

File: src/languages/es.ts
Comment (pecanoro) [inline src/languages/es.ts:1834]:
```suggestion
        subtitle: 'Utiliza Expensify Travel para obtener las mejores ofertas de viaje y gestionar todos los gastos de tu negocio en un solo lugar.',
```

File: src/languages/es.ts
Comment (pecanoro) [inline src/languages/es.ts:1842]:
```suggestion
            title: 'Por favor, lee los Términos y condiciones para reservar viajes',
```

File: src/languages/es.ts
Comment (pecanoro) [inline src/languages/es.ts:1843]:
```suggestion
            subtitle: 'Para permitir la opción de reservar viajes en su espacio de trabajo debe aceptar nuestros ',
```

File: src/languages/es.ts
Comment (pecanoro) [inline src/languages/es.ts:1846]:
```suggestion
            helpDocIntro: 'Consulta este ',
```

File: src/languages/es.ts
Comment (pecanoro) [inline src/languages/es.ts:1847]:
```suggestion
            helpDocOutro: 'para obtener más información o comunícate con Concierge o tu gestor de cuentas.',
```

**Review by pecanoro — CHANGES_REQUESTED** [review]

File: src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx
Comment (pecanoro) [inline src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx:50]: Move this comment above the if, please

```suggestion
        // When we are redirected to the Settings tab from the OldDot, we don't want to call the Welcome.show() method.
        // To prevent this, the value of the bottomTabRoute?.name is checked here
        if (
            Boolean(currentRoute && currentRoute.name !== NAVIGATORS.BOTTOM_TAB_NAVIGATOR && currentRoute.name !== NAVIGATORS.CENTRAL_PANE_NAVIGATOR) ||
```

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Comment by shawnborton** [comment]
On iOS, the bottom button feels awfully close to the home/drag bar. Do we have the proper SafeSpace below the button there? 
CleanShot 2024-04-29 at 10 50 45@2x

**Comment by dubielzyk-expensify** [comment]
The illustration looks like it doesn't have much padding either:
CleanShot 2024-04-30 at 15 29 41@2x

I believe the height of that card illustration area should be `260px` and it should have about `28px` padding top and bottom. But we could round up to `32` or `20`

CleanShot 2024-04-30 at 15 31 55@2x

**Comment by twisterdotcom** [comment]
Bump on these requested changes @cdOut

**Comment by WojtekBoman** [comment]
@dubielzyk-expensify @shawnborton Could you send me this illustration with exact dimensions in .lottie format on Slack?

> The illustration looks like it doesn't have much padding either: CleanShot 2024-04-30 at 15 29 41@2x
> 
> I believe the height of that card illustration area should be `260px` and it should have about `28px` padding top and bottom. But we could round up to `32` or `20`
> 
> CleanShot 2024-04-30 at 15 31 55@2x

**Comment by WojtekBoman** [comment]
> The illustration looks like it doesn't have much padding either: CleanShot 2024-04-30 at 15 29 41@2x
> 
> I believe the height of that card illustration area should be `260px` and it should have about `28px` padding top and bottom. But we could round up to `32` or `20`
> 
> CleanShot 2024-04-30 at 15 31 55@2x

@dubielzyk-expensify @shawnborton Could you send me on Slack this illustration in .lottie format with exact dimensions?

**Comment by shawnborton** [comment]
> It should behave the same as on other pages in RHP with a button at the bottom of the screen. Does this button have a different style on pages like "Assign a Task" or "Save the World > I Know a Teacher"?

@WojtekBoman check out the Save button on the Display name page - notice how there is a bit more space below the button?
CleanShot 2024-04-30 at 07 33 06@2x

**Comment by WojtekBoman** [comment]
@shawnborton @dubielzyk-expensify UI Issues have been fixed :) Could you check if it looks ok now?

**Comment by shawnborton** [comment]
Will fire up a test build for us, but let me know if you have updated screenshots in the original comment as well!

**Comment by WojtekBoman** [comment]
> Will fire up a test build for us, but let me know if you have updated screenshots in the original comment as well!

I can't update the description of this PR because I'm not the author :(

**Comment by shawnborton** [comment]
Bummer, but feel free to just make a new comment with the updated screenshots at least!

**Comment by shawnborton** [comment]
What's the best way to test this using the links above? I added myself to the spotnanaTravel beta but I'm not seeing the Book Travel option show up. Any ideas @twisterdotcom?

**Comment by WojtekBoman** [comment]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>


Screenshot 2024-04-30 at 18 36 27
Screenshot 2024-04-30 at 18 36 05

</details>

<details>
<summary>iOS: Native</summary>


Screenshot 2024-04-30 at 18 42 43
Screenshot 2024-04-30 at 18 42 32

</details>

<details>
<summary>iOS: mWeb Safari</summary>


Screenshot 2024-04-30 at 18 43 04
Screenshot 2024-04-30 at 18 42 59

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


Screenshot 2024-04-30 at 18 28 41
Screenshot 2024-04-30 at 18 28 35

</details>

<details>
<summary>MacOS: Desktop</summary>


Screenshot 2024-04-30 at 18 32 45
Screenshot 2024-04-30 at 18 32 41

</details>

I got an error when building `Android/Native`, so I haven't attached screenshots for this platform

**Comment by twisterdotcom** [comment]
@shawnborton you gotta wait for the beta cache to flush before your account is added. I do see it for pecan@expensivepie.com: 
image

File: src/languages/en.ts
Comment (twisterdotcom) [inline src/languages/en.ts:1842]: This should be "realtime" instead of "real time". cc @jamesdeanexpensify. (Spanish can remain as it is).

File: src/pages/Travel/TravelTerms.tsx
Comment (twisterdotcom) [inline src/pages/Travel/TravelTerms.tsx:43]: Should be https://use.expensify.com/travelterms

File: src/pages/Travel/TravelTerms.tsx
Comment (twisterdotcom) [inline src/pages/Travel/TravelTerms.tsx:68]: Should be https://use.expensify.com/travelterms

File: src/pages/Travel/TravelTerms.tsx
Comment (twisterdotcom) [inline src/pages/Travel/TravelTerms.tsx:72]: @bfitzexpensify do we have a help article ready, even if it's empty?

**Comment by twisterdotcom** [comment]
Are these links correct? 
image

On oldDot the terms go to https://use.expensify.com/travelterms, so we should use that.

Looking at it now - should we link "terms & conditions" twice? Or maybe we only need to link it at the bottom with the checkbox, as we've done on oldDot:
image

I tried web, desktop and Android and those were my only comments.

File: src/pages/Travel/TravelTerms.tsx
Comment (bfitzexpensify) [reply src/pages/Travel/TravelTerms.tsx:72]: No, but I've added a note in https://github.com/Expensify/Expensify/issues/383227 to make a PR to update this once the Help Docs are live on ExpensifyHelp

**Comment by shawnborton** [comment]
Hmm I still can't seem to get the Book travel option to show up in my global create:
CleanShot 2024-05-01 at 08 13 43@2x

But the screenshots above look good to me.

**Comment by rushatgabhane** [comment]
Hello everyone 👋 
I've created a new PR because i can't add commits to this one. I've applied the suggestions above. 

Please review when you can - https://github.com/Expensify/App/pull/41493

**Pull request merged**
